[feat]:SignIn and SignUp functionality

ultronjs opened the pull request:
Integrated SignIn and SIngUp API and all the functionality related to the same.
Adding to Wishlist and Cart only if user is logged in.
Toast on Error and Info also Added

ultronjs committed:
[feat]:SignIn and SignUp functionality

Integrated SignIn and SIngUp API and all the functionality related to the same.
Adding to Wishlist and Cart only if user is logged in.
Toast on Error and Info also Added

ultronjs committed:
update:Updating SignIn Route in SignUp Page

JantuDeb commented on src/components/LoginForm.jsx:
```diff
@@ -1,8 +1,41 @@
+  const { signInStatusDispatch } = useAuth();
+  const onChangeHandler = (e) => {
+    setSignIn((prevState) => ({
+      ...prevState,
+      [e.target.name]: e.target.value,
```
Liked that yo are using this dynamic  key

JantuDeb commented on src/components/Toast.jsx:
```diff
@@ -0,0 +1,24 @@
+import React from 'react'
+
+function Toast({type,msg}) {
+    const variantOfToast = {
+      Success: ["toast_success", "fa-shield-check"],
+      Info: ["toast_info", "fa-info-circle"],
+      Warning: ["toast_warning", "fa-exclamation-circle"],
+      Error: ["toast_error", "fa-times-circle"],
+    };
+  return (
+    <div class="toast_wrapper toast_rightside my-x-small">
+      <div class={`toast ${variantOfToast[type][0]}`}>
+        <i class={`far ${variantOfToast[type][1]} fa-2x toast_icon`}></i>
+        <div class="toast_content">
+          <span class="toast_heading">{type}</span>
+          <span class="toast_body">{msg}</span>
+        </div>
+        <i class="far fa-times-circle fa-lg toast_dismiss"></i>
+      </div>
+    </div>
+  );
+}
```
Liked the way you are using toast

JantuDeb commented on src/utils/axios.js:
```diff
@@ -1,19 +1,28 @@
-  headers: {'Accept': '*/*',authorization: encodedToken}
+  headers: { Accept: "*/*", authorization: localStorage.getItem("token") },
+});
+privateInstance.interceptors.request.use(async (req) => {
+  console.log("intercerper", req);
+  return req;
+});
+privateInstance.interceptors.response.use(async (res) => {
+  console.log("intercerper response", res);
+  return res;
```
Remove interceptors if you are not doing anything.

ultronjs replied:
Done

ultronjs committed:
[update]:removing interceptors

removing interceptors .Not required anymore.
git Comment Resolve

Priya-Patel31 commented on src/context/wishListContext.js:
```diff
@@ -1,26 +1,27 @@
         const {status,data} = await privateInstance
       .get("/user/wishlist")
       if(status ===200){
         wishListDispatch({ type: "SET_DATA", payload: data.wishlist })
```
Suggestion : here you can use payload: {wishlist : data.wishlist} that would be more understandable that which kind of data it is .

ultronjs replied:
Done

Priya-Patel31 commented on src/index.js:
```diff
@@ -4,22 +4,26 @@ import "./index.css";
-        </WishListProvider>
-      </CartProvider>
-    </FilterProvider>
+    <ToastProvider>
```
You can have a separate file for these contexts and then add import that single component here .

Priya-Patel31 commented on src/reducers/filterReducer.js:
```diff
@@ -25,7 +24,6 @@ export const filterReducer = (state, action) => {
     case "sortByAccessoriesType":
       return { ...state, sortByAccessoriesType: action.payload };
     case "sortByBrands":
```
Use uppercase for the name of these cases .

Priya-Patel31 commented on src/components/Toast.jsx:
```diff
@@ -0,0 +1,24 @@
+import React from 'react'
+
+function Toast({type,msg}) {
```
Really like the way you have made the dynamic toasts.

Priya-Patel31 commented on src/components/SignUpForm.jsx:
```diff
@@ -1,59 +1,116 @@
+      <div className="form_container">
+        <div className="form sign_up">
+          <div className="form_img">
+            <img src="/assets/login_logo.png" alt="" />
```
give some text in the alt tag.

ultronjs replied:
Done

Priya-Patel31 reviewed (approved):
The code is well formatted and really liked the toast which you have added . Also , liked the emptyCart and emptyWishlist components which you have added in your code . Just have few suggestions and minor changes you can go through them.

ultronjs committed:
[fix]:Added Search Functionality and Fixed some bugs

-Added Search Functionality
-Added Guest Login
-Added 404 page
-Fixed the navigation on logout
-Show password fixed on signIn and signUp page

ultronjs committed:
[fix]:adding alt text

Resolving Git comments
